blog/tests/views_test.py

```diff
@@ -0,0 +1,179 @@
+
+        authorization = self.client.login(username=self.USERNAME, password=self.PASSWORD)
+        self.assertTrue(authorization)
+        response = self.client.get(reverse('post_publish', kwargs={'pk': post.pk}), follow=True)
+        self.assertRedirects(response, reverse('post_detail', kwargs={'pk': post.pk}))
+        response = self.client.get(reverse('post_list'), follow=True)
+        self.assertContains(response, post)
+
+    def test_delete_post(self):
+        """Testing deleting post: before and after login; post deleting."""
+        response = self.client.get(reverse('post_remove', kwargs={'pk': 1}))
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username=self.USERNAME, password=self.PASSWORD)
+        self.assertTrue(authorization)
+        response = self.client.get(reverse('post_remove', kwargs={'pk': 1}))
+        self.assertEqual(404, response.status_code)
+        post = Post.objects.create(author=self.user, title='Tst', text='superText')
+        other_post = Post.objects.create(author=self.user, title='Test_other', text='Other text')
+        response = self.client.get(reverse('post_remove', kwargs={'pk': post.pk}), follow=True)
+        self.assertRedirects(response, reverse('post_list'))
+        response = self.client.get(reverse('post_draft_list'), follow=True)
+        self.assertNotContains(response, post)
+        self.assertContains(response, other_post)
+
+    def test_add_comment(self):
+        """Testing adding comment to post."""
+        self.post = Post.objects.create(author=self.user, title='Test', text='superText')
+        response = self.client.get(reverse('add_comment', kwargs={'pk': self.post.pk}))
+        self.assertEqual(200, response.status_code)
+        response = self.client.post(reverse('add_comment', kwargs={'pk': self.post.pk}),
+                                    {'author': self.user, 'text': 'Super'}, follow=True)
+        self.assertRedirects(response, reverse('post_detail', kwargs={'pk': self.post.pk}))
+
+    def test_comment_approve(self):
+        """Testing comment approve view: comment visibility only after approving."""
+        self.post = Post.objects.create(author=self.user, title='Test', text='superText')
+        self.comment = Comment.objects.create(post=self.post, author=self.user, text='superComment')
+        response = self.client.get(reverse('post_detail', kwargs={'pk': self.post.pk}), follow=True)
+        self.assertNotContains(response, self.comment)
+        authorization = self.client.login(username=self.USERNAME, password=self.PASSWORD)
+        self.assertTrue(authorization)
+        response = self.client.get(reverse('comment_approve', kwargs={'pk': self.comment.pk}), follow=True)
+        self.assertRedirects(response, reverse('post_detail', kwargs={'pk': self.post.pk}))
+        response = self.client.get(reverse('post_detail', kwargs={'pk': self.post.pk}), follow=True)
+        self.client.logout()
```

Comment:
In this case you have all data send before logout(and also get response)
you need to sent one more request to server this time and then see if there is that post